R/jira.R

```diff
@@ -8,103 +8,392 @@
+    description = list(description),
+    name = list(name)
+  )
+
+  return(resolution)
+}
+
+#' Create Status
+#'
+#' Creates a status with information
+#'
+#' @param self_url URL of API endpoint
+#' @param description Description of status
+#' @param iconUrl URL to icon associated with status
+#' @param name Name of status
+#' @param id ID associated with status
+#' @param statusCategory_self URL to status category API endpoint
+#' @param statusCategory_id ID associated with status category
+#' @param statusCategory_key key associated with status category
+#' @param statusCategory_colorName name of color associated with status category
+#' @param statusCategory_name name of status category
+#' @return A list named 'status' containing status's information
+#' @export
+create_status <- function(self_url = "https://domain.org/jira/rest/api/2/status/6",
+                          description = "The issue is considered finished, the resolution is correct. Issues which are not closed can be reopened.",
+                          iconUrl = "https://domain.org/jira/images/icons/statuses/closed.png",
+                          name = "Closed",
+                          id = "6",
+                          statusCategory_self = "https://domain.org/jira/rest/api/2/statuscategory/3",
+                          statusCategory_id = 3,
+                          statusCategory_key = "done",
+                          statusCategory_colorName = "green",
+                          statusCategory_name = "Done") {
+  status <- list(
+    self = list(self_url),
+    description = list(description),
+    iconUrl = list(iconUrl),
+    name = name,
+    id = id,
+    statusCategory = list(
+      self = list(statusCategory_self),
+      id = list(statusCategory_id),
+      key = list(statusCategory_key),
+      colorName = list(statusCategory_colorName),
+      name = list(statusCategory_name)
+    )
+  )
+
+  return(status)
+}
+
+#' Create ext_info_cell for issues (with comments)
+#'
+#' Creates the ext_info_cell with issues with comments
+#'
+#' @return ext_info_cell with comment fields
+#' @export
+# values can be changed either here if hard coded or in respective create function below
+create_ext_info_issues_comments <- function() {
+  ext_info_cell <- list(
+    summary = list("Issue Description"),
+    issuetype = create_issuetype(),
+    components = create_components(component_details),
+    creator = create_creator(),
+    created = list("2007-07-08T06:07:06.000+0000"),
+    description = list(),
+    reporter = create_reporter(),
+    resolution = create_resolution(self_url = "https://domain.org/jira/rest/api/2/resolution/1"),
+    resolutiondate = list("2007-08-13T19:12:33.000+0000"),
+    assignee = list(),
+    updated = list("2008-05-12T08:01:39.000+0000"),
+    status = create_status(),
+    comment = create_comment(comment_details)
+  )
+
+  return(ext_info_cell)
+}
+
+#' Create Comment
+#'
+#' Creates comment field with information
```

Comment:
https://support.atlassian.com/jira-software-cloud/docs/watch-share-and-comment-on-an-issue/